src/qtism/runtime/tests/Route.php

```diff
@@ -1185,4 +1186,33 @@ public function getRouteItemPosition(RouteItem $routeItem): int
             throw new OutOfBoundsException($msg);
         }
     }
+
+    public function getEffectiveBranchRules(): BranchRuleCollection
```

Comment:
I think this method should be on the `RouteItem`, not here

Reply:
We are using some methods from `Route`, like:
* getRouteItemsByAssessmentSection
* getCurrentTestPartRouteItems
* etc

Reply:
Do you need them? Why not get the `$routeItem->getTestPart()->getBranchRules()` and `$routeItem->getAssessmentSection()->getBranchRules()` ?

Reply:
I need to detect if the item is the last one in test part or in section.
If the item is not the last one - I should not use test part/section rules.

Reply:
Moved to `RouteItem` with code changes